File: enaFrontend/src/app/models/dossier.ts
import { Arrete } from "./arrete";
import { BulletinSalaire } from "./bulletin-salaire";
import { Casier } from "./casier";
import { CertificatAdministratif } from "./certificat-administratif";
import { CertificatNationalite } from "./certificat-nationalite";
import { Diplome } from "./diplome";
import { Engagement } from "./engagement";
import { Memoire } from "./memoire";
import { ValidationDossier } from "./validation-dossier";
import { Visite } from "./visite";

export class Dossier {
    public id!: number;
    public eleve!: number;
    public voieAcces!: any;
    public arretes!: Arrete[];
    public bulletins!: BulletinSalaire[];
    public casiers!: Casier[];
    public certificatAdministratifs!: CertificatAdministratif[];
    public certificatNationalites!: CertificatNationalite[];
    public diplomes!: Diplome[];
    public engagements!: Engagement[];
    public memoires!: Memoire[];
    public validationDossiers!: ValidationDossier[];
    public visites!: Visite[];
}

File: ena/src/main/java/com/ena/entities/Dossier.java
package com.ena.entities;

import com.ena.deserializer.DossierDeserializer;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@JsonDeserialize(using = DossierDeserializer.class)
public class Dossier implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JsonBackReference
    private Eleve eleve;
    @ManyToOne
    private VoieAcces voieAcces;
    @OneToMany(mappedBy = "dossier")
    private List<Arrete> arretes;
    @OneToMany(mappedBy = "dossier")
    private List<BulletinSalaire> bulletinSalaires;
    @OneToMany(mappedBy = "dossier")
    private List<Casier> casiers;
    @OneToMany(mappedBy = "dossier")
    private List<CertificatAdministratif> certificatAdministratifs;
    @OneToMany(mappedBy = "dossier")
    private List<CertificatNationalite> certificatNationalites;
    @OneToMany(mappedBy = "dossier")
    private List<Diplome> diplomes;
    @OneToMany(mappedBy = "dossier")
    private List<Engagement> engagements;
    @OneToMany(mappedBy = "dossier")
    private List<Memoire> memoires;
    @OneToMany(mappedBy = "dossier")
    private List<ValidationDossier> validationDossiers;
    @OneToMany(mappedBy = "dossier")
    private List<Visite> visites;
}
